### PR TITLE
Capitalisation: Add camelCase

### LINKED ISSUE #5768: add camel case to extended_capitalisation_policy
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

Currently, following the implementation of #5470, the options for `extended_capitalisation_policy` are `['consistent', 'upper', 'lower', 'pascal', 'capitalise', 'snake']`, where the `'pascal'` option specifies that the first character should be uppercase. Please add a `'camel'` option for camel case, to specify that the first character should be lowercase.

### Use case

Currently, I must set `extended_capitalisation_policy` to `None` under `[sqlfluff:rules:capitalisation.functions]`, to avoid SQLFluff making unwanted changes to my camelCase function names. As a result, I can't benefit from SQLFluff's ability to enforce consistency in this area. Having a `'camel'` option would restore this benefit, and give SQLFluff [all three of the most common naming conventions](https://khalilstemmler.com/blogs/camel-case-snake-case-pascal-case/#The-most-common-naming-conventions).

### Dialect

This should be relevant to multiple dialects.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by WittierDinosaur**
### Brief summary of the change made
This PR adds camelCase to the extended capiltalisation policy. Added tests to cover its functionality.
Fixes #5768 

### Are there any other side effects of this change that we should be aware of?
I made it so the complex types are all only available when explicitly set in the config.
All 3 are either destructive to the code, or could allow poorly linted files through. Most people would use upper or lower anyway, so requiring the complex cases to be explicit should be a non-issue.

This will require a minor version bump.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/core/rules/config_info.py [hunk 1]
```diff
@@ -60,10 +60,18 @@
...
+            "capitalise",
+            "snake",
+            "camel",
+        ],
         "definition": (
-            "The capitalisation policy to enforce, extended with PascalCase "
-            "and snake_case. "
+            "The capitalisation policy to enforce, extended with PascalCase, "
+            "snake_case, and camelCase. "
             "This is separate from ``capitalisation_policy`` as it should not be "
```
Comment (alanmcruickshank) [inline src/sqlfluff/core/rules/config_info.py:75]: I think the precedence rules here aren't very easy to guess. I think from running through the examples below, that I understand the logic - but I think we should document it here.

If nothing else - someone will ask on the slack channel and I'd like to be able to point them to something to explain _why_ the default config is acting the way that it is.

**Review by alanmcruickshank — CHANGES_REQUESTED** [review]
Can you elaborate in the docs? I _think_ the logic makes sense but I'm having trouble checking that it's consistent without it clearly documenting what it's _supposed_ to do.

**Comment by nstringham** [comment]
Any update on this?

**Comment by atishay** [comment]
How can we help getting this merged?

**Review by alanmcruickshank — APPROVED** [review]
LGTM. I agree there's some strange inference on fixing camel case situations, but I don't have any better ideas!

**Pull request merged**